### ISSUE TITLE
[FSDP] Support scheduled learning rate

### ISSUE THREAD

**Issue by PopSoda2002** [THUDM/slime#962]
Currently FSDP only supports fixed learning, does not support like warm up, cosine decay like megatron

**Comment by ChangyiYang** [comment]
I would glad to take it with @aaaandychen :)

**Comment by ChangyiYang** [comment]
After some investigation, we have two options for LR scheduling:

A. Use PyTorch built-in schedulers

📌 https://github.com/pytorch/pytorch/blob/main/torch/optim/lr_scheduler.py

Pros: easy integration, maintained by PyTorch
Cons: no warmup+decay in one place → not fully compatible with Megatron

B. Implement a custom scheduler based on Megatron

📌 https://github.com/NVIDIA/Megatron-LM/blob/e847643f88758b7f5577b119240d3dfd1b4b4661/megatron/core/optimizer_param_scheduler.py

Pros: supports all Megatron features (warmup, min_lr, cosine/linear/inv-sqrt/WSD) 
Cons: needs own maintenance, only support lr, not weight decay

Pytorch currently does not suport dynamic weight decay.

I would prefer option B, any opinion is welcomed!

**Comment by PopSoda2002** [comment]
Fell free to implement whatever you want! But I am not sure we need decouple FSDP and Megatron

**Comment by ChangyiYang** [comment]
I will go with option B. Since we are copying the code from megatron, not refer from it, FSDP and Megatron are still decoupled.

**Comment by PopSoda2002** [comment]
solved by https://github.com/THUDM/slime/pull/1040 thanks! @ChangyiYang
